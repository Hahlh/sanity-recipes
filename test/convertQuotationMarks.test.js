const convertQuotationMarks = require('../snippets/convertQuotationMarks')
const singleQuotationMarks = require('./fixtures/singleQuotationMarks.fixture')
const doubleQuotationMarks = require('./fixtures/doubleQuotationMarks.fixture')

test('converts quotation marks to guillemets by default', () => {
  expect(convertQuotationMarks(doubleQuotationMarks)).toMatchSnapshot()
})

test('can converts quotation marks to specific characters', () => {
  expect(convertQuotationMarks(doubleQuotationMarks, {open: '“', close: '”'})).toMatchSnapshot()
})

test('can converts single quotation marks', () => {
  expect(convertQuotationMarks(singleQuotationMarks, {find: "'"})).toMatchSnapshot()
})

test('can converts single quotation marks to specific characters', () => {
  expect(convertQuotationMarks(singleQuotationMarks, {find: "'", open: '"', close: '"'})).toEqual(
    doubleQuotationMarks
  )
})
