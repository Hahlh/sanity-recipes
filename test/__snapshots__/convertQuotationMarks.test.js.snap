// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can converts quotation marks to specific characters 1`] = `
Array [
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [],
        "text": "Warning: “",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Sanity”",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": " is",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "em",
        ],
        "text": " “addictive”.",
      },
    ],
  },
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Unmatched quotation marks? “",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "sure, why not?",
      },
    ],
  },
]
`;

exports[`can converts single quotation marks 1`] = `
Array [
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [],
        "text": "Warning: «",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Sanity»",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": " is",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "em",
        ],
        "text": " «addictive».",
      },
    ],
  },
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Unmatched quotation marks? «",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "sure, why not?",
      },
    ],
  },
]
`;

exports[`converts quotation marks to guillemets by default 1`] = `
Array [
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [],
        "text": "Warning: «",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Sanity»",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": " is",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "em",
        ],
        "text": " «addictive».",
      },
    ],
  },
  Object {
    "_type": "block",
    "children": Array [
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "Unmatched quotation marks? «",
      },
      Object {
        "_type": "span",
        "marks": Array [
          "strong",
        ],
        "text": "sure, why not?",
      },
    ],
  },
]
`;
